src/script/calling/CallingRepository.ts

```diff
@@ -1165,11 +1168,37 @@ export class CallingRepository {
     this.wCall?.requestVideoStreams(this.wUser, convId, VSTREAMS.LIST, JSON.stringify(payload));
   }
 
+  readonly showCallQualityFeedbackModal = (): void => {
+    if (!this.selfUser) {
+      return;
+    }
+
+    const {setQualityFeedbackModalShown} = useCallAlertState.getState();
+
+    try {
+      const qualityFeedbackStorage = localStorage.getItem(CALL_QUALITY_FEEDBACK_KEY);
+      const currentStorageData = qualityFeedbackStorage ? JSON.parse(qualityFeedbackStorage) : {};
+      const currentUserDate = currentStorageData?.[this.selfUser.id];
+      const currentDate = new Date().toISOString();
+
+      if (typeof currentUserDate === 'undefined' || (currentUserDate !== null && currentDate >= currentUserDate)) {
```

Comment:
Do we need typeof here?
```suggestion
      if (currentUserDate === undefined || (currentUserDate !== null && currentDate >= currentUserDate)) {
```

Reply:
`currentDate >= currentUserDate`
We are comparing two strings here, would be better if we compare date or unicode timestamps.